src/datumaro/plugins/data_formats/tabular.py

```diff
@@ -0,0 +1,207 @@
+# Copyright (C) 2021-2023 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import errno
+import os
+import os.path as osp
+from typing import Dict, List, Optional, Tuple, Type, Union
+
+from datumaro.components.annotation import AnnotationType, Categories, Tabular, TabularCategories
+from datumaro.components.dataset_base import DatasetBase, DatasetItem
+from datumaro.components.errors import MediaTypeError
+from datumaro.components.exporter import Exporter
+from datumaro.components.importer import ImportContext, Importer
+from datumaro.components.media import Table, TableDtype, TableRow
+from datumaro.util.os_util import find_files
+
+# Only supports '.csv' extention.
+TABULAR_EXTENSIONS = [
+    "csv",
+]
+
+
+class TabularDataBase(DatasetBase):
+    NAME = "tabular"
+    """
+    Compose a tabular dataset.
+    """
+
+    def __init__(
+        self,
+        path: str,
+        *,
+        target: Optional[Union[str, List[str]]] = None,
+        dtype: Optional[Dict[str, Type[TableDtype]]] = None,
+        ctx: Optional[ImportContext] = None,
+    ) -> None:
+        """
+        Read a tabular dataset.
+
+        Args:
+            path (str) : Path to a tabular dataset. (csv file or folder contains csv files).
+            target (optional, str or list) : Target column or list of target columns.
+                If this is not specified (None), the last column is regarded as a target column.
+                In case of a dataset with no targets, give an empty list as a parameter.
+            dtype (optional, dict (str: str)) : Dictionay of column name -> type (str, int, or float).
+                This can be used when automatic type inferencing is failed.
+        """
+        paths: List[str] = list()
```

Comment:
```suggestion
        paths: List[str] = []
```
[] is more efficient than list().
See https://stackoverflow.com/questions/5790860/and-vs-list-and-dict-which-is-better

Reply:
Good to know. Thanks!